### PR TITLE
Add preview to RGB stereo node

### PR DESCRIPTION
**Pull request by roni-kreinin**
Added parameters:
- Enable/disable video stream
- Enable/disable preview stream
- Set color camera resolution
- Set preview resolution
- Enable/disable urdf

Example use case:
200x200 preview:
`ros2 launch depthai_examples rgb_stereo_node.launch.py useVideo:=False usePreview:=True previewWidth:=200 previewHeight:=200 colorResolution:=1080p`

4K video:
`ros2 launch depthai_examples rgb_stereo_node.launch.py  colorResolution:=4K`

Disable URDF:
`ros2 launch depthai_examples rgb_stereo_node.launch.py publish_urdf:=False`

Notes:
I get errors if I put BridgePublisher inside `if(useVideo)` when useVideo is false, would appreciate help with this so that the topics are not advertised for streams that are disabled.

### CODE REVIEW

**Comment by saching13** [comment]
Thanks Roni. Can you make the changes discussed in discord by making them pointers to avoid errors ?
> I get errors if I put BridgePublisher inside if(useVideo) when useVideo is false, would appreciate help with this so that the topics are not advertised for streams that are disabled.

**Comment by roni-kreinin** [comment]
Working on it.

**Comment by roni-kreinin** [comment]
Do you want me to add publishers for the left/right mono images to this node? I think it would be nice to have the option.

**Comment by saching13** [comment]
> Do you want me to add publishers for the left/right mono images to this node? I think it would be nice to have the option.

Why not just use the `stereo_inertial_node` ? It does RGB publishing, RGB Stereo and option to change to Left-right already

**Comment by roni-kreinin** [comment]
That is true. Maybe it would be good to combine some of these nodes into a single main node that is configurable. That way users can decide which streams they want published, what resolution, whether to launch rviz/pointcloud/urdf, and so on.

**Comment by saching13** [comment]
> That is true. Maybe it would be good to combine some of these nodes into a single main node that is configurable. That way users can decide which streams they want published, what resolution, whether to launch rviz/pointcloud/urdf, and so on.

Agreed. Hence most of the work is being done on `stero_inertial_node` since it has most stereo based required features.

File: depthai_examples/ros2_src/rgb_stereo_node.cpp [hunk 1]
```diff
@@ -139,23 +203,52 @@ int main(int argc, char** argv){
...
+        rgbPreviewPublish = std::make_unique<dai::rosBridge::BridgePublisher<sensor_msgs::msg::Image, dai::ImgFrame>>(previewQueue,
+                                                                            node, 
+                                                                            std::string("color/preview"),
+                                                                            std::bind(&dai::rosBridge::ImageConverter::toRosMsg, 
+                                                                            &rgbConverter, // since the converter has the same frame name
+                                                                                            // and image type is also same we can reuse it
+                                                                            std::placeholders::_1, 
+                                                                            std::placeholders::_2) , 
+                                                                            30,
+                                                                            rgbCameraInfo,
```
Comment (saching13) [inline depthai_examples/ros2_src/rgb_stereo_node.cpp]: You need to change this for preview size here.

**Review by saching13 — APPROVED** [review]
Except that one comment. LGTM. 
Feel free to merge it after adding a different info for that.

**Pull request merged**